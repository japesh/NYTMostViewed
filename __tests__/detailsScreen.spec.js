import 'react-native';
import React from 'react';
import DetailScreen from '../pages/detailScreen';
import renderer from 'react-test-renderer';

it('Detail screen test case', () => {
    const details = {
        title: "my title",
        abstract: "My abstract"
    }
    const navigation = { getParam: (name) => details[name], navigate: jest.fn() };
    const tree = renderer.create(<DetailScreen navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
    const {title} = DetailScreen.navigationOptions({navigation});
    expect(title).toBe("Details my title")
});