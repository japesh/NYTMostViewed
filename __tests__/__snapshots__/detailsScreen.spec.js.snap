// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail screen test case 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#454545",
        "paddingBottom": 10,
      }
    }
  >
    my title
  </Text>
  <Text>
    My abstract
  </Text>
</View>
`;
